src/Main.cpp

```diff
@@ -63,96 +69,74 @@ void LockMouse(sf::RenderWindow& window) {
 void UnlockMouse(sf::RenderWindow& window) {
   window.setMouseCursorVisible(true);
 }
-void extractDLL(const char* fname, int res_id) {
-  //Get the current path of the executable
-  HMODULE hModule = GetModuleHandleW(NULL);
-  CHAR path[MAX_PATH];
-  GetModuleFileName(hModule, path, MAX_PATH);
-
-  //Replace the exe filename with the dll
-  std::string str(path);
-  size_t last_dir = str.find_last_of('\\');
-  str = str.substr(0, last_dir + 1) + std::string(fname);
-
-  //Copy the dll to the path
-  const Res res_dll(res_id);
-  std::ofstream fout(str, std::ios::binary);
-  fout.write((const char*)res_dll.ptr, res_dll.size);
-  fout.close();
-}
 
-#ifdef _DEBUG
 int main(int argc, char *argv[]) {
-#else
-int WinMain(HINSTANCE hInstance, HINSTANCE, LPTSTR lpCmdLine, int nCmdShow) {
-#endif
-  //Extract openAL dll
-  extractDLL("openal32.dll", IDR_OPENAL);
-
   //Make sure shader is supported
   if (!sf::Shader::isAvailable()) {
-    MessageBox(nullptr, TEXT("Graphics card does not support shaders"), TEXT("ERROR"), MB_OK);
+    std::cout << "Graphics card does not support shaders" << std::endl;
```

Comment:
This was more of a quick and dirty solution since there is no cross-platform message box API.

IMO either an exception should be used or a custom dialog window through `SFML`.

Reply:
I've bookmarked a couple of APIs to try later: tinyfiledialogs and nativefiledialog. They might be useful here.